### PR TITLE
runtime, blob: make ErrorAs panic on nil or non-pointer target

### PR DESCRIPTION
**Pull request by jba**
Fixes # 1201.

### CODE REVIEW

File: blob/blob.go [hunk 1]
```diff
@@ -420,6 +421,9 @@ func (b *Bucket) As(i interface{}) bool {
 // ErrorAs converts i to provider-specific types.
 // See Bucket.As for more details.
```
Comment (vangent) [inline blob/blob.go]: Add the comment here about it panicking if i is nil or not a pointer.

File: runtimevar/runtimevar.go [hunk 2]
```diff
@@ -180,9 +180,12 @@ func (w *wrappedError) Error() string {
 // ErrorAs converts i to provider-specific types.
 // See Snapshot.As for more details.
 func ErrorAs(err error, i interface{}) bool {
```
Comment (vangent) [inline runtimevar/runtimevar.go]: And here.

File: blob/blob.go (hunk 1 above)
Comment (jba) [reply blob/blob.go]: Done.

File: runtimevar/runtimevar.go (hunk 2 above)
Comment (jba) [reply runtimevar/runtimevar.go]: Done.

**Pull request merged**